sublime_jedi/go_to.py

```diff
@@ -151,11 +157,144 @@ def prepare_option(self, option):
 
 class SublimeJediFindUsages(BaseLookUpJediCommand, sublime_plugin.TextCommand):
     """
-    Find object usages
+    Find object usages, and optionally rename objects.
     """
     def run(self, edit):
-        ask_daemon(self.view, self._window_quick_panel_open_window, 'usages')
+        self.edit = edit
+        ask_daemon(self.view, self.handle_usages, 'usages')
+
+    def handle_usages(self, view, options) -> None:
+        if not options:
+            return
+
+        active_window = view.window()
+
+        # remember filenames
+        self.options = options
+
+        # remember current file location
+        self.point = self.view.sel()[0]
+
+        # expands selection to all of "focused" symbol
+        name = expand_selection(self.view, self.point)
+
+        def handle_rename(new_name: str) -> None:
+            groups = []  # type: List[List[Tuple[str, int, int]]]
+            files = set()  # type: Set[str]
+
+            for option in options:
+                file = option[0]
+                if not file:  # can't replace text (or even show usages) in unsaved file
+                    continue
+                if file in files:
+                    groups[-1].append(option)
+                else:
+                    groups.append([option])
+                files.add(file)
+
+            for group in groups:
+                rename_in_file(group, group[0][0], new_name)
+
+        def rename_in_file(group, file, new_name):
+            # type: (List[Tuple[str, int, int]], str, str) -> None
+            with open(file) as f:
+                text = f.read()
+            original_text = text
+            offset = 0
+
+            for option in group:
+                assert text and name
+                _, row, col = option
+                point = text_point(original_text, row-1, col-1)
+
+                text = text[:point + offset] + new_name + text[point + offset + len(name):]
+                offset += len(new_name) - len(name)
+
+            with open(file, "w") as f:
+                f.write(text)
+
+        def handle_choose(idx):
+            if not name:
+                return
+            if idx == 0:
+                view.window().show_input_panel("New name:", name, handle_rename, None, None)
+                return
+            self._jump_to_in_window(idx - 1 if idx != -1 else idx)
+
+        def handle_highlight(idx):
+            if idx == 0:
+                return
+            partial(self._jump_to_in_window, transient=True)(idx - 1 if idx != -1 else idx)
+
+        # Show the user a selection of filenames
+        files = set()  # type: Set[str]
+        for option in options:
+            files.add(option[0])
```

Comment:
could be looks like `files = {option[0] for option in options}`

Reply:
Good call, the comprehension is much nicer